pkg/testing/runner/debian.go

```diff
@@ -207,6 +207,41 @@ func (DebianRunner) Run(ctx context.Context, verbose bool, sshClient *ssh.Client
 	return result, nil
 }
 
+// Diagnostics gathers any diagnostics from the host.
+func (DebianRunner) Diagnostics(ctx context.Context, c *ssh.Client, logger Logger, destination string) error {
+	// take ownership, as sudo tests will create with root permissions (allow to fail in the case it doesn't exist)
+	diagnosticDir := "$HOME/agent/build/diagnostics"
+	_, _, _ = sshRunCommand(ctx, c, "sudo", []string{"chown", "-R", "$USER:$USER", diagnosticDir}, nil)
+	stdOut, _, err := sshRunCommand(ctx, c, "ls", []string{"-1", diagnosticDir}, nil)
+	if err != nil {
+		//nolint:nilerr // failed to list the directory, probably don't have any diagnostics (do nothing)
+		return nil
+	}
+	eachDiagnostic := strings.Split(string(stdOut), "\n")
+	for _, filename := range eachDiagnostic {
+		filename = strings.TrimSpace(filename)
+		if filename == "" {
+			continue
+		}
+
+		// don't use filepath.Join as we need this to work in Windows as well
```

Comment:
Is the problem here that filepath.Join will format this for the host system instead of the remote system?

On first read this confused me since filepath.Join is supposed be cross-platform.

Reply:
Can you improve the comments to they state the problem you are avoiding directly?

Reply:
I improved the comment.